test/e2e/specs/editor/plugins/block-hooks.spec.js

```diff
@@ -84,9 +84,15 @@ test.describe( 'Block Hooks API', () => {
 					page.locator( '.entry-content > *' )
 				).toHaveClass( [
 					'wp-block-heading',
-					getHookedBlockClassName( 'after', 'core/heading' ),
-					'dummy-paragraph',
-					getHookedBlockClassName( 'last_child', blockType ),
+					`${ getHookedBlockClassName(
+						'after',
+						'core/heading'
+					) } wp-block-paragraph`,
+					'wp-block-paragraph',
+					`${ getHookedBlockClassName(
+						'last_child',
+						blockType
+					) } wp-block-paragraph`,
```

Comment:
Here is the slightly confusing part of the refactored test.  This test checks that the hooked block has the correct class name on the front end. If a Paragraph block is hooked, the `.wp-block-paragraph` class name is also required in addition to the class name generated by `getHookedBlockClassName()`.

cc @ockham @gziolo